core/src/test/java/org/apache/calcite/test/TypeCoercionTest.java

```diff
@@ -545,16 +545,16 @@ assert equals(result1, expected)
     expr("'12.3'/cast(5 as double)")
         .columnType("DOUBLE NOT NULL");
     expr("'12.3'/5.1")
-        .columnType("DECIMAL(19, 18) NOT NULL");
+        .columnType("DECIMAL(19, 8) NOT NULL");
     expr("12.3/'5.1'")
-        .columnType("DECIMAL(19, 0) NOT NULL");
+        .columnType("DECIMAL(19, 8) NOT NULL");
     // test binary arithmetic with two strings.
     expr("'12.3' + '5'")
-        .columnType("DECIMAL(19, 19) NOT NULL");
+        .columnType("DECIMAL(19, 9) NOT NULL");
     expr("'12.3' - '5'")
-        .columnType("DECIMAL(19, 19) NOT NULL");
+        .columnType("DECIMAL(19, 9) NOT NULL");
     expr("'12.3' * '5'")
-        .columnType("DECIMAL(19, 19) NOT NULL");
+        .columnType("DECIMAL(19, 18) NOT NULL");
```

Comment:
here comes the 18
![image](https://user-images.githubusercontent.com/15643702/95651199-310ead00-0b1b-11eb-9f98-a118b4daf239.png)